sdk/core/core/src/az_http_response.c

```diff
@@ -258,3 +259,12 @@ AZ_NODISCARD az_result az_http_response_get_body(az_http_response* response, az_
   response->_internal.parser.next_kind = _az_HTTP_RESPONSE_KIND_EOF;
   return AZ_OK;
 }
+
+void _az_http_response_reset(az_http_response* http_response)
```

Comment:
Where do we call this function? I don't see any call sites in this PR. Shouldn't the samples call it?

Reply:
it is called by the transport policy. 
This is an internal method that customer does not know about it and gets never exposed to them